test_sitemap_utils.py

```
@@ -0,0 +1,237 @@
+                    </url>
+                </urlset>
+                """
+                
+        mock_make_request.side_effect = side_effect
+        
+        # Enable respect_robots_txt
+        self.parser.respect_robots_txt = True
+        urls = self.parser.parse_sitemap("https://example.com")
+        
+        # Check that we found the URL from the custom sitemap
+        self.assertEqual(len(urls), 1)
```

Ruff (0.8.2)

Use a regular `assert` instead of unittest-style `assertEqual`

Replace `assertEqual(...)` with `assert ...`

(PT009)